File: frontend/src/types/sermon.ts
import type { BaseEntity } from './common';

export interface Sermon extends BaseEntity {
  title: string;
  slug: string;
  description: string;
  content: string;
  preacher: string;
  preacherImage: string;
  date: string;
  duration: string;
  category: SermonCategory;
  tags: string[];
  thumbnailUrl: string;
  videoUrl?: string;
  audioUrl?: string;
  scriptureReferences: ScriptureReference[];
  isFeatured: boolean;
  viewCount: number;
  likeCount: number;
}

export interface ScriptureReference {
  book: string;
  chapter: number;
  verseStart: number;
  verseEnd?: number;
  text: string;
}

export type SermonCategory =
  | 'faith'
  | 'love'
  | 'prayer'
  | 'worship'
  | 'family'
  | 'leadership'
  | 'healing'
  | 'prophecy'
  | 'salvation'
  | 'holy-spirit';

File: backend/FirstLoveChurch.API/Models/Sermons/Sermon.cs
using FirstLoveChurch.API.Models.Common;
using FirstLoveChurch.API.Models.Identity;

namespace FirstLoveChurch.API.Models.Sermons;

public class Sermon : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Content { get; set; }
    public string? VideoUrl { get; set; }
    public string? AudioUrl { get; set; }
    public string? ImageUrl { get; set; }
    public string? ScriptureReferences { get; set; }
    public bool IsFeatured { get; set; }
    public int ViewCount { get; set; }
    public int LikeCount { get; set; }
    public string? Slug { get; set; }

    public Guid? SeriesId { get; set; }
    public SermonSeries? Series { get; set; }

    public Guid? CategoryId { get; set; }
    public SermonCategory? Category { get; set; }

    public string? SpeakerId { get; set; }
    public ApplicationUser? Speaker { get; set; }

    public ICollection<SermonTagMapping> TagMappings { get; set; } = new List<SermonTagMapping>();
    public ICollection<SermonComment> Comments { get; set; } = new List<SermonComment>();
}
